File: src/types/Vehicle.ts
export type Vehicle = {
  registrationNumber: string;
  make: string;
  model: string;
  colour: string;
  fuelType: string;
  engineCapacity: number;
  yearOfManufacture: number;
  vehicleAge: string;
  wheelplan: string;
  dateOfLastV5CIssued: string;
  typeApproval: string;
  co2Emissions: number;
  registrationPlace: string;
  tax: {
    taxStatus: string;
    taxDueDate: string;
    days: string;
  };
  mot: {
    motStatus: string;
    motDueDate: string;
    days: number;
  };
};

File: QuoteJourney.API/Data/Entities/Vehicle.cs
namespace QuoteJourney.API.Data.Entities;

public class Vehicle
{
    public string RegistrationNumber { get; set; }
    public string Make { get; set; }
    public string Model { get; set; }
    public string Colour { get; set; }
    public string FuelType { get; set; }
    public int EngineCapacity { get; set; }
    public int YearOfManufacture { get; set; }
    public string VehicleAge { get; set; }
    public string Wheelplan { get; set; }
    public DateTime DateOfLastV5CIssued { get; set; }
    public string TypeApproval { get; set; }
    public int Co2Emissions { get; set; }
    public string RegistrationPlace { get; set; }
    public Tax Tax { get; set; }
    public Mot Mot { get; set; }
}

public class Tax
{
    public string TaxStatus { get; set; }
    public DateTime TaxDueDate { get; set; }
    public string Days { get; set; }
}

public class Mot
{
    public string MotStatus { get; set; }
    public DateTime MotDueDate { get; set; }
    public int Days { get; set; }
}
